GitUI/Editor/FileViewerInternal.cs

```diff
@@ -69,6 +73,56 @@ public FileViewerInternal()
             VRulerPosition = AppSettings.DiffVerticalRulerPosition;
         }
 
+        private void SelectionManagerSelectionChanged(object sender, EventArgs e)
+        {
+            string text = TextEditor.ActiveTextAreaControl.TextArea.SelectionManager.SelectedText;
+            TextEditor.Document.MarkerStrategy.RemoveAll(m => true);
+
+            IList<TextMarker> selectionMarkers = GetTextMarkersMatchingWord(text);
+
+            foreach (var selectionMarker in selectionMarkers)
+            {
+                TextEditor.Document.MarkerStrategy.AddMarker(selectionMarker);
+            }
+
+            _diffHighlightService.AddPatchHighlighting(TextEditor.Document);
+            TextEditor.ActiveTextAreaControl.TextArea.Invalidate();
+        }
+
+        /// <summary>
+        /// Create a list of TextMarker instances in the Document that match the given text
+        /// </summary>
+        /// <param name="word">The text to match.</param>
+        [NotNull]
+        private IList<TextMarker> GetTextMarkersMatchingWord(string word)
+        {
+            var selectionMarkers = new List<TextMarker>();
+            if (word.IsNullOrWhiteSpace())
+            {
+                return selectionMarkers;
```

Comment:
Can we make this 
```suggestion
                return Array.Empty<TextMarker>();
```
and move the `var selectionMarkers = new List<TextMarker>();` down?

Reply:
Done